src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -0,0 +1,343 @@
+              " Consider using larger size than 10KBytes");
+        }
+
+        File vidMapJSONFile = (vidMapJSONFileName.equals(DEFAULT_VIDMAP_FILE_NAME)) ?
+            new File(workspaceDir + "/" + vidMapJSONFileName) :
+            new File(vidMapJSONFileName);
+
+        File callsetMapJSONFile = (callsetMapJSONFileName.equals(DEFAULT_CALLSETMAP_FILE_NAME)) ?
+            new File (workspaceDir + "/" + callsetMapJSONFileName) :
+            new File(callsetMapJSONFileName);
+
+        logger.info("Vid Map JSON file will be written to " + vidMapJSONFile);
+        logger.info("Callset Map JSON file will be written to " + callsetMapJSONFile);
+        
+        long variantContextBufferSize = vcfBufferSizePerSample * sampleToVCFMap.size();
+
+        logger.info("Writing data to array - " + workspace + "/" + arrayName);
+
+        try {
+            logger.info("GenomicsDB import batch started");
+            importer = new GenomicsDBImporter(sampleToVCFMap, mergedHeader, intervals.get(0), workspace,
+                                              arrayName, variantContextBufferSize, segmentSize,
+                                              vidMapJSONFile.getAbsolutePath(),
+                                              callsetMapJSONFile.getAbsolutePath());
+        } catch (IOException e) {
+            throw new UserException("GenomicsDB constructor has errors: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Input argument "overwriteExistingWorkspace" defaults to false.
+     * The tool should create a new workspace by default if the workspace
+     * doesn't already exist, but only overwrite an existing workspace
+     * if the user specifies this argument.
+     *
+     * @return  The workspace directory
+     */
+    private File checkOrCreateWorkspace() {
+        File workspaceDir = new File(workspace);
+
+        if (overwriteExistingWorkspace) {
+            IOUtils.tryDelete(workspaceDir);
+        }
+
+        int ret = GenomicsDBImporter.createTileDBWorkspace(workspaceDir.getAbsolutePath());
+
+        if (ret > 0) {
+            File tempFile = new File(workspaceDir.getAbsolutePath() + "/__tiledb_workspace.tdb");
+            if (!tempFile.exists()) {
+                throw new UserException(workspaceDir.getAbsolutePath() + " is not a valid GenomicsDB workspace");
+            }
+
+            logger.info("Importing data to GenomicsDB workspace: " + workspaceDir);
+        } else if (ret < 0) {
+            throw new UserException("Error creating GenomicsDB workspace: " + workspaceDir);
+        }
+
+        return workspaceDir;
+    }
+
+    /**
+     * A complete traversal from start to finish. This method will import all samples
+     * specified in the input GVCF files.
+     */
+    @Override
+    public void traverse() {
+        try {
+            importer.importBatch();
+        } catch (IOException e) {
+            throw new UserException("GenomicsDB importBatch raised exception: " + e.getMessage());
```

Comment:
Don't just pass in `e.getMessage() to the `UserException` constructor -- pass in `e` itself, so that it will appear in the stack trace. Eg.,

```
throw new UserException("Error during GenomicsDB import", e);
```